tsdb/chunkenc/xor.go

```diff
@@ -253,6 +253,19 @@ type xorIterator struct {
 	err    error
 }
 
+func (it *xorIterator) Seek(t int64) bool {
+	if it.err != nil {
+		return false
+	}
+
+	for t > it.t || it.numRead == 0 {
```

Comment:
Shouldn't logic be opposite here? We only want to advance, if `t < it.t`. Looks like unit test here would be a good idea.

Reply:
There is already: https://github.com/prometheus/prometheus/blob/07b7090dd78fafafbe6c229f77fc469d4bca4fd0/tsdb/chunkenc/chunk_test.go#L91

But I double check this.

Reply:
so `t > it.t` is fine as  we only want to advance if required t is smaller than iterator t (:

Reply:
When I applied your suggestion, test failed , double checked if test makes sesne as well.